include/extractor/serialization.hpp

```diff
@@ -94,6 +98,54 @@ inline void write(storage::io::FileWriter &writer,
     storage::serialization::write(writer, node_data_container.name_ids);
     storage::serialization::write(writer, node_data_container.travel_modes);
 }
+
+// read/write for conditional turn restrictions file
+inline void read(const boost::filesystem::path &path, std::vector<TurnRestriction> &restrictions)
```

Comment:
Instead of taking a path this should take an instance of `FileReader` like the other functions in this file.

You should also add a function `readRestrictions` in `files.hpp` that takes a path and then calls this function.

We do this to abstract away the notion of files from parsing the data.